aodntools/timeseries_products/velocity_aggregated_timeseries.py

```diff
@@ -0,0 +1,316 @@
+    INSTRUMENT = ds.createDimension('INSTRUMENT', size=n_files)
+
+    UCUR = ds.createVariable(varname='UCUR', datatype='float', zlib=True, dimensions=('OBSERVATION'), fill_value=999999.0)
+    VCUR = ds.createVariable(varname='VCUR', datatype='float', zlib=True, dimensions=('OBSERVATION'), fill_value=999999.0)
+    WCUR = ds.createVariable(varname='WCUR', datatype='float', zlib=True, dimensions=('OBSERVATION'), fill_value=999999.0)
+    DEPTH = ds.createVariable(varname='DEPTH', datatype='float', zlib=True, dimensions=('OBSERVATION'), fill_value=999999.0)
+    UCURqc = ds.createVariable(varname='UCUR_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    VCURqc = ds.createVariable(varname='VCUR_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    WCURqc = ds.createVariable(varname='WCUR_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    DEPTHqc = ds.createVariable(varname='DEPTH_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    TIME = ds.createVariable(varname='TIME', datatype='double', zlib=True, dimensions=('OBSERVATION'))
+    instrument_index = ds.createVariable(varname='instrument_index', datatype='int', zlib=True, dimensions=('OBSERVATION'))
+
+    source_file = ds.createVariable(varname='source_file', datatype='S256', dimensions=('INSTRUMENT'))
+    instrument_id = ds.createVariable(varname='instrument_id', datatype='S256', dimensions=('INSTRUMENT'))
+    LATITUDE = ds.createVariable(varname='LATITUDE', datatype='float', dimensions=("INSTRUMENT"))
+    LONGITUDE = ds.createVariable(varname='LONGITUDE', datatype='float', dimensions=("INSTRUMENT"))
+    NOMINAL_DEPTH = ds.createVariable(varname='NOMINAL_DEPTH', datatype='float', dimensions=('INSTRUMENT'))
+
+    ## main loop
+    for index, file in enumerate(files_to_agg):
+        print(index)
+        with xr.open_dataset(file) as nc:
+
+            ## in water data only
+            nc = in_water(nc)
+
+            start = sum(varlen_list[:index + 1])
+            end = sum(varlen_list[:index + 2])
+            n_cells = get_nvalues(nc)[1]
+            UCUR[start:end] = get_varvalues(nc, 'UCUR')
+            UCURqc[start:end] = get_varvalues(nc, 'UCUR_quality_control')
+            VCUR[start:end] = get_varvalues(nc, 'VCUR')
+            VCURqc[start:end] = get_varvalues(nc, 'VCUR_quality_control')
+            if 'WCUR' in nc.data_vars:
+                WCUR[start:end] = get_varvalues(nc, 'WCUR')
+                WCURqc[start:end] = get_varvalues(nc, 'WCUR_quality_control')
+            else:
+                WCUR[start:end] = np.full(varlen_list[index], np.nan)
+                WCURqc[start:end] = np.full(varlen_list[index], np.nan)
+            ##calculate depth
+            if 'HEIGHT_ABOVE_SENSOR' in nc.dims:
+                DEPTH[start:end] = (nc.DEPTH + nc.HEIGHT_ABOVE_SENSOR.T).values.flatten()
```

Comment:
`HEIGHT_ABOVE_SENSOR` is in the opposite direction to `DEPTH`, so it should be `DEPTH - HEIGHT_ABOVE_SENSOR`!

Also, I don't quite understand the use of the `.T` attribute. I'm assuming it's meant to be a transpose, but `HEIGHT_ABOVE_SENSOR` is a 1-dimensional array. ([Xarray](http://xarray.pydata.org/en/stable/generated/xarray.DataArray.html#xarray.DataArray) doesn't seem to provide any documentation of this attribute, while [numpy](https://numpy.org/doc/1.16/reference/generated/numpy.ndarray.T.html#numpy.ndarray.T) says it just return the original array if ndim < 2)

Reply:
Yes, you're correct. The .T has no effect as HEIGHT_ABOVE_SENSOR is 1D.fixed